### PR TITLE
Fix: Added retry mechanism for fetchBalance

### LINKED ISSUE #839: add retry mechanism for fetch balance
Retry mechanism isn't implemented for FetchBalance function. Needs to be done.

### PR DESCRIPTION
**Pull request by shekhar2807**
# Description

- Added retry mechanism for fetchBalance
- Modified tests for fetchBalance
 
Fixes #839

### CODE REVIEW

File: utils/common.go [hunk 1]
```diff
@@ -26,10 +27,26 @@ func (*UtilsStruct) ConnectToClient(provider string) *ethclient.Client {
...
-	opts := UtilsInterface.GetOptions()
-	return CoinInterface.BalanceOf(coinContract, &opts, address)
+	var (
+		balance *big.Int
+		err     error
+	)
+	err = retry.Do(
+		func() error {
+			address := common.HexToAddress(accountAddress)
+			coinContract := UtilsInterface.GetTokenManager(client)
```
Comment (ashish10677) [inline utils/common.go:37]: rename this as erc20Contract

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: utils/common.go (hunk 1 above)
Comment (shekhar2807) [reply utils/common.go:37]: There is a separate PR/Issue for this #857 @ashish10677 . If needed to be done here, let me know.

Comment (ashish10677) [reply utils/common.go:37]: This is a new change you're adding so you can do it here.

Comment (shekhar2807) [reply utils/common.go]: Done @ashish10677

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**